### PR TITLE
JaegerExporter: Send a batch for each remote service

### PR DESCRIPTION
**Pull request by CodeBlanch**
This PR is similar to #483. Jaeger will associate all spans in a batch to the ServiceName sent for that batch. This means calls to external services that aren't instrumented (ex: sql, redis, APIs, etc.) won't be detected as dependencies. This really reduces the usefulness of the tool.

What this PR does is update the JaegerExporter to set "peer.service" (same fuzzy logic as #483) on spans and then use that to send batches per service to the Jaeger agent. Should work out of the box for the existing OpenTelemetry collectors (and the SQL one in PR).

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerConversionExtensions.cs [hunk 1]
```diff
@@ -43,13 +43,46 @@ public static class JaegerConversionExtensions
...
+            var jaegerTags = new List<JaegerTag>();
+
+            Tuple<string, int> peerService = null;
+            foreach (var label in span.Attributes)
+            {
+                var tag = label.ToJaegerTag();
+
+                if (tag.VStr != null
+                    && PeerServiceKeyResolutionDictionary.TryGetValue(label.Key, out int priority)
+                    && (peerService == null || peerService.Item2 > priority))
```
Comment (MikeGoldsmith) [inline src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerConversionExtensions.cs]: I think this should be `<` as the highest priority is in ascending order from 0 - we're looking for the lowest priority.

**Comment by MikeGoldsmith** [comment]
Change looks good, just one query regarding selecting peer.service based on priority ranking.

File: src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerConversionExtensions.cs (hunk 1 above)
Comment (CodeBlanch) [reply src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerConversionExtensions.cs]: I switched them around so it reads better.

**Comment by CodeBlanch** [comment]
My original implementation actually broke the size calculations, I fixed that up. In doing so I hit the same deadlock as #492. I fixed it on here as well so my test would pass. If that PR goes first, I'll port the fix over. But I like mine better :D Everyone loves their own kids, right?

**Comment by MikeGoldsmith** [comment]
@SergeyKanzhelev this looks good to review / merge.

**Pull request merged**